wp-includes/rest-api/endpoints/class-wp-rest-customize-panels-controller.php

```diff
@@ -0,0 +1,201 @@
+<?php
+/**
+ * REST API: WP_REST_Customize_Panels_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since ?.?.?
+ */
+
+/**
+ * Core class to access posts via the REST API.
+ *
+ * @since ?.?.?
+ *
+ * @see WP_REST_Controller
+ */
+class WP_REST_Customize_Panels_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 *
+	 * @since 4.7.0
+	 * @access public
+	 */
+	public function __construct() {
+		$this->namespace = 'customize/v1';
+		$this->rest_base = 'panels';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 *
+	 * @since ?.?.?
+	 * @access public
+	 *
+	 * @see register_rest_route()
+	 */
+	public function register_routes() {
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base, array(
+			array(
+				'methods'             => WP_REST_Server::READABLE,
+				'callback'            => array( $this, 'get_items' ),
+				'permission_callback' => array( $this, 'get_items_permissions_check' ),
+				'args'                => $this->get_collection_params(),
+			),
+			'schema' => array( $this, 'get_public_item_schema' ),
+		) );
+
+		register_rest_route( $this->namespace, '/' . $this->rest_base . '/(?P<panel>[\w-]+)', array(
```

Comment:
A panel may have `[` and `]` characters in it. I don't think I have seen it, but I can envision where it may happen. At least, for sections there are ids like `posts[page]` and `posts[product]`.